src/context/data-context.tsx

```diff
@@ -0,0 +1,63 @@
+import { createContext, useContext, useEffect, useState } from "react";
+import { collection, query, where, getDocs } from "firebase/firestore";
+
+import { childrenType, NoteType } from "./context.type";
+import { db } from "../config/firebase-config";
+
+type DataContextProviderProps = {
+  children: childrenType;
+};
+
+type DataContextType = {
+  notes: NoteType[];
+  labels: string[];
+  isLoading: boolean;
+  getNotes: () => Promise<void>;
+  setNotes: React.Dispatch<React.SetStateAction<NoteType[]>>;
+};
+
+const DataContext = createContext({} as DataContextType);
+
+const DataContextProvider = ({ children }: DataContextProviderProps) => {
+  const [notes, setNotes] = useState<NoteType[]>([]);
+  const [labels, setLabels] = useState<string[]>([]);
+  const [isLoading, setIsloading] = useState(false);
+  const uid = localStorage.getItem("userId");
+
+  useEffect(() => {
+    getNotes();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
remove comment

Reply:
The comment is to disable the ESLint rules.